src/server/pps/server/api_server.go

```diff
@@ -1332,9 +1335,9 @@ func (a *apiServer) GetLogs(request *pps.GetLogsRequest, apiGetLogsServer pps.AP
 			}
 			statsCommit = jobPtr.StatsCommit
 			pipelineInfo, err = a.inspectPipeline(pachClient, jobPtr.Pipeline.Name)
-		}
-		if err != nil {
-			return fmt.Errorf("could not get pipeline information for %s: %v", request.Pipeline.Name, err)
```

Comment:
This line would segfault if the error came from the `request.Job != nil` branch.